src/utils/hooks/use-redirect-location.js

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+import { useLocation, matchPath } from 'react-router-dom';
+import { routes, params } from '../../routes.config';
+
+/**
+ * to trigger different actions based on the current location in the flowchart
+ * and this hook is only called when the page is reload
+ */
+export const useRedirectLocationInFlowchart = (
+  modularPipelinesTree,
+  nodes,
+  onLoadNodeData,
+  onToggleFocusMode,
+  onToggleModularPipelineActive,
+  onToggleModularPipelineExpanded,
+  onUpdateActivePipeline,
+  reload
+) => {
+  const { pathname, search } = useLocation();
+
+  const activePipelineId = search.substring(
+    search.indexOf(params.pipeline) + params.pipeline.length,
+    search.indexOf('&')
+  );
+
+  const decodedPipelineId = decodeURI(activePipelineId);
+
+  const matchedSelectedNode = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.flowchart.selectedNode],
+  });
+
+  const matchedFocusedNode = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.flowchart.focusedNode],
+  });
+
+  useMemo(() => {
+    if (matchedSelectedNode && Object.keys(nodes).length > 0) {
+      const nodeId = search.split(params.selected)[1];
+
+      const modularPipeline = nodes[nodeId];
+      const hasModularPipeline = modularPipeline.length > 0;
+
+      if (hasModularPipeline) {
+        onToggleModularPipelineExpanded(modularPipeline);
+        onUpdateActivePipeline(decodedPipelineId);
+        onLoadNodeData(nodeId);
+      } else {
+        onLoadNodeData(nodeId);
+        onUpdateActivePipeline(decodedPipelineId);
+      }
+    }
+
+    if (matchedFocusedNode && Object.keys(modularPipelinesTree).length > 0) {
+      const modularPipelineId = search.split(params.focused)[1];
+      const modularPipeline = modularPipelinesTree[modularPipelineId];
+
+      onToggleFocusMode(modularPipeline.data);
+      onToggleModularPipelineActive(modularPipelineId, true);
+      onUpdateActivePipeline(decodedPipelineId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [reload]);
```

Comment:
To my previous comment about using the browser navigation forward/back buttons: if you add `search` to the dependency array it helps, though it isn't perfect.

Reply:
yeah good point let me try that

Reply:
hey so my update for this issue include: 
- pass "search" to the dependency array so it can track when it changed
- update value null to both `onLoadNodeData` and `onToggleFocusMode` when the URL is matched the main page
```
if (matchedFlowchartMainPage) {
      onLoadNodeData(null);
      onToggleFocusMode(null);
    }
```
- ensure we also set `onToggleFocusMode(null)` when selectedNode path is matched, and the other way round for when focusedNode path is matched
```
// Reset the focus mode to null when when using the navigation buttons
      onToggleFocusMode(null);


// Reset the node data to null when when using the navigation buttons
      onLoadNodeData(null);
```